### DISCUSSION TITLE
Docs Restructuring

### DISCUSSION THREAD

**Discussion by sbalnojan** [meltano/meltano#6738] [General]
I feel the documentation for Meltano could benefit from a restructuring, and I would love to hear community comments on this one because the goal is:

1. to get Meltano first timers onboarded as quickly as possible onto their first productionized pipeline
2. to get advanced users help quickly, or at least find "holes" in the docs. 

I think the diataxis framework is a great way of doing this, here's a rough sketch of how this could be used even better:

![image](https://user-images.githubusercontent.com/43072233/189638543-e817736b-649f-4def-b35b-4d282d92da1f.png)

The general idea would be to:

1. Introduce a "Getting Started section". In the diataxis framework this would be a true tutorial aimed at first timers. It would source a lot from the current "Getting Started" but cut down into much smaller pieces. Some content would be taken out (basically everything that "explains" or "provides alternatives"). The idea is to get people running fast.
2. Introduce a "How to" section with specific answers to specific problems. So while in the Getting Started section you will learn how to set up one example of an ELT process, in the how to, you should have a guide that will set up **any** ELT process. 
3. Introduce more references, the place where I just want a list of all commands & arguments. 
4. Introduce "advanced tutorials" - I think this section will be a key section for goal (2), as Meltano really really thrives in the advanced section.

How: This is going to involve a lot of shuffling around of content, and then a lot of hole plugging. The "How to" section will require a lot of new content. 

Any questions, ideas, comments (contributors? :-D)? 

FWIW, I already kicked off parts of this with two PRs, https://github.com/meltano/meltano/pull/6737 & https://github.com/meltano/meltano/pull/6709.

**Comment by visch** [comment]
- Search on the main meltano site should either index docs, or make it very clear you need to go do docs.meltano.com to get the docs for Meltano
- How to guide idea: "How to find out if it's the tap, target, or meltano". (Ideally meltano would make this easier over time as well, but right now this takes some significant know how in some situations!)

Great you're looking into this! :D

**Reply by sbalnojan** [reply]
> * Search on the main meltano site should either index docs, or make it very clear you need to go do docs.meltano.com to get the docs for Meltano

both great points. As we're building out Managed, Managed docs will also have a place somewhere so we definitely need to make the separation very clear to get people to the right place quickly.

**Comment by sbalnojan** [comment]
From Slack: "_tutorials coupled with youtube videos are incredibly helpful (like dbt X jaffle shop)"_

**Comment by sbalnojan** [comment]
Getting started section starts to take shape: https://docs.meltano.com/getting-started/meltano-at-a-glance
